### ISSUE TITLE
`AppNavigationItem` counter should be visible on active

### ISSUE THREAD

**Issue by raimund-schluessler** [nextcloud/nextcloud-vue#3094]
The counter slot in the `AppNavigationItem` component is now hidden (or replaced with the Actions menu) when the `AppNavigationItem` is hovered or active.

While hiding it on hover, might be ok (but debatable), I think it is not correct to hide the counter when the item is active. For the Tasks app this would mean that the counter is not visible for the task list currently viewed. This is not intuitive and should be changed, I think. This is a regression due to #3048.

See here:
![AppNavigationItem](https://user-images.githubusercontent.com/2496460/186367851-a85508d1-2d38-42c2-abac-90ec36e6b177.gif)

**Comment by marcoambrosini** [comment]
It should indeed be visible when the element is active @raimund-schluessler, and that should be the case actually. Is it behaving differently? What you're showing in the clip is not the active state

**Comment by raimund-schluessler** [comment]
Hm, indeed. Let me double check with an actual app.

**Comment by raimund-schluessler** [comment]
@marcoambrosini So I checked with the Tasks app again and there are multiple problems, sadly. The counter is not shown when the entry is active, see here:
![Screenshot 2022-08-24 at 20-54-22 Aufgaben - Nextcloud](https://user-images.githubusercontent.com/2496460/186500338-e3476c3a-3dbd-443f-9997-c9fae1f09968.png)

I don't understand why yet, to be honest. I have disabled all custom adjustments, but it still happens.

Furthermore, the `app-navigation-entry__utils` fixed width of `44px` breaks a use case of the Tasks and Calendar app. We also use the counter slot to show the sharing icon and the avatar, which are to be shown permanently. With a fixed width of `44px` the sharing icon pushes the counter of the item.

**Comment by raimund-schluessler** [comment]
So this width:
https://github.com/nextcloud/nextcloud-vue/blob/e28a7178ca938e7175c1c78ee29055f59cdffaef/src/components/NcAppNavigationItem/NcAppNavigationItem.vue#L781
creates issues. It prevents showing `Actions` and `Counter` at the same time.

**Comment by raimund-schluessler** [comment]
And hiding the counter slot content on hover creates the issue that whatever you put into this slot cannot be interactive, since it's hidden when you try to click it. This also breaks the use case of the Calendar and Tasks of putting the sharing and avatar items into the counter slot.

I would like to hear what @ChristophWurst and @skjnldsv think about this. Although the counter slot is by name not meant to contain anything else than a counter, it worked quite well for both apps. So it the counter is to be hidden, we need a new slot for these elements.

**Comment by ChristophWurst** [comment]
Yes, this is also problematic in Mail where we conditionally show a longer text for the counter: https://github.com/nextcloud/mail/pull/7071#issuecomment-1224445390. And this info should show also for active elements.

**Comment by raimund-schluessler** [comment]
@marcoambrosini So I double checked with the Taks app and this is how it behaves at the moment:
1. The actions are *only shown on hover*, not when the item is active (`displayActionsOnHoverFocus` does not account for the active state of the item). This makes it impossible to click the actions on mobile.
2. The `.app-navigation-entry__utils` are to narrow, because their width is limited to 44 px, which cuts off the actions, counter and/or additional elements put in the counter slot
3. The counter is hidden on hover, which makes it impossible to click interactive elements (e.g. Sharing button) in the counter slot.

These are all regressions of #3061.

I would say the solutions for the problems are as follows:
1. Was agreed upon above. The actions need to be visible if the AppNavigationItem is active.
2. The `.app-navigation-entry__utils` need to be as wide as necessary. I think this is also clear, since Calendar, Tasks and Mail use it like this.
3. The counter should be always visible if the AppNavigationItem is active. Not seeing the counter for the active item seems weird and it at least allows to click interactive elements in the counter slot. I would further prefer to not hide the counter on hover, even if the item is not active, but that's up for debate I guess.

 @jancborchardt @nimishavijay A bit of feedback here would be good, because the `AppNavigationItem` is broken for Tasks, Calendar and Mail in the current state.

**Comment by raimund-schluessler** [comment]
Undo and editing are also broken after #3061. This PR really created quite some fallout :see_no_evil:
